static/js/components/SiteContentEditor.tsx

```diff
@@ -4,60 +4,90 @@ import { useMutation, useRequest } from "redux-query-react"
 import { useSelector } from "react-redux"
 import { FormikHelpers } from "formik"
 
+import SiteContentForm from "./forms/SiteContentForm"
+
 import {
+  createWebsiteContentMutation,
   editWebsiteContentMutation,
   EditWebsiteContentPayload,
+  NewWebsiteContentPayload,
   websiteContentDetailRequest
 } from "../query-configs/websites"
 import { getWebsiteContentDetailCursor } from "../selectors/websites"
-import SiteEditContentForm from "./forms/SiteEditContentForm"
-import { contentFormValuesToPayload } from "../lib/site_content"
+import {
+  contentFormValuesToPayload,
+  splitFieldsIntoColumns
+} from "../lib/site_content"
 import { getResponseBodyError, isErrorResponse } from "../lib/util"
 
 import { EditableConfigItem, Website } from "../types/websites"
-
-type SiteFormValues = {
-  [key: string]: string
-}
+import { ContentFormType, SiteFormValues } from "../types/forms"
 
 interface Props {
-  uuid: string
+  uuid: string | null
   visibility: boolean
   toggleVisibility: () => void
   site: Website
   configItem: EditableConfigItem
+  formType: ContentFormType
+  contentType: string
+  repeatListingRequest: () => void
```

Comment:
Can we change this name to `contentListingRequest` or `websiteContentListingRequest` here and in `SiteContentListing.tsx`?